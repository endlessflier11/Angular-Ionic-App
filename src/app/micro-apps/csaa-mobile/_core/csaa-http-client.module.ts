import { NgModule } from '@angular/core';
import { AngularHttpAdapter } from './services/http/adapters/angular-http.adapter';
import { HttpClient, HttpClientModule } from '@angular/common/http';
import { IonicHttpAdapter } from './services/http/adapters/ionic-http.adapter';
import { HTTP } from '@ionic-native/http/ngx';
import { UNI_HTTP_ADAPTER } from './services/http/uni-http.model';

@NgModule({
  imports: [HttpClientModule],
  providers: [
    HTTP,
    { provide: UNI_HTTP_ADAPTER, multi: true, useClass: AngularHttpAdapter, deps: [HttpClient] },
    {
      provide: UNI_HTTP_ADAPTER,
      multi: true,
      useClass: IonicHttpAdapter,
      deps: [HTTP],
    },
  ],
  exports: [HttpClientModule],
})
export class CsaaHttpClientModule {}
