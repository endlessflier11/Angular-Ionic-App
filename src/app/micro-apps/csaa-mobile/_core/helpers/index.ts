export * from './router.helpers';
export * from './misc.helpers';
export * from './async.helper';
export * from './error.helper';
export * from './string.helper';
