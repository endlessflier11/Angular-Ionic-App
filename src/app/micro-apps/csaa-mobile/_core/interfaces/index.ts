export * from './auth.interface';
export * from './router';
export * from './http.interface';
export * from './vehicle.interface';
export * from './ionic-lifecycle.interface';
export * from './analytics.interface';
export * from './policy.interface';
export * from './payment.interface';
export * from './config.interface';
export * from './ngxs-store.interface';
