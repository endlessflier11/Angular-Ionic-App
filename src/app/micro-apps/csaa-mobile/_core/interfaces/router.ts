import { Route, UrlTree } from '@angular/router';
import { NavigationOptions } from '@ionic/angular/providers/nav-controller';
import { Observable } from 'rxjs';

interface AppRoute extends Route {
  name?: string;
  isGroupIndex?: boolean; // If true, it means route is the index of parent. Then it inherits all parent properties
}

export type AppRoutes = AppRoute[];

export interface RouteParam {
  [key: string]: string;
}

export interface NamedRouteConfig {
  name: string;
  path: string;
  children?: NamedRouteConfig[];
}

export interface NamedRoute {
  name: string;
  path: string;
  relativePath: string;
}

export interface CanComponentDeactivate {
  canDeactivate: () =>
    | Observable<boolean | UrlTree>
    | Promise<boolean | UrlTree>
    | boolean
    | UrlTree;
}

export interface CsaaRouteStackItem {
  name: string;
  params?: RouteParam;
  options?: NavigationOptions;
}
