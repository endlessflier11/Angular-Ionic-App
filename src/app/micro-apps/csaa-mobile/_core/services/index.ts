export * from './router/router.service';
export * from './config/config.service';
export * from './auth/auth.service';
export * from './sso/sso.service';
export * from './storage/storage.service';
export * from './http/http.service';
export * from './analytics.service';
export * from './policy.service';
export * from './payment.service';
export * from './global-state.service';
export * from './call.service';
export * from './metadata.service';
export * from './pdf-display.service';
export * from './webview/webview.service';
