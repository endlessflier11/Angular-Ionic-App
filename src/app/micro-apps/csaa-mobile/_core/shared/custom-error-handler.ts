import { ErrorHandler, Injectable } from '@angular/core';
import { AlertController } from '@ionic/angular';

import { CsaaCoreModule } from '../../csaa-core/csaa-core.module';

export class RefreshTokenExpiredError extends Error {
  constructor(...params) {
    super(...params);
    // Set the prototype explicitly.
    Object.setPrototypeOf(this, RefreshTokenExpiredError.prototype);
  }
}
export class UnknownRefreshError extends Error {
  constructor(...params) {
    super(...params);
    // Set the prototype explicitly.
    Object.setPrototypeOf(this, UnknownRefreshError.prototype);
  }
}

@Injectable({
  providedIn: CsaaCoreModule,
})
export class CustomErrorHandler implements ErrorHandler {
  constructor(
    private alertCtrl: AlertController // TODO v5: keep?  private signOut: SignOutDirective
  ) {}

  handleError(error: any) {
    const httpErrorCode = error.httpErrorCode;
    console.error(error);
    switch (httpErrorCode) {
      // todo: ... do we care to give a different message for each error?
      default:
        this.showError(error);
    }
  }

  async getAlertForError(error) {
    if (error instanceof RefreshTokenExpiredError) {
      return await this.alertCtrl.create({
        header: `Login expired`,
        subHeader: `Please sign up again.`,
        buttons: [
          {
            text: 'Okay',
            role: 'cancel',
            handler: () => 
              // TODO v5: keep?
              // this.signOut.signOut();
               true
            ,
          },
        ],
      });
    }

    if (error instanceof UnknownRefreshError) {
      return this.alertCtrl.create({
        header: 'Something went wrong',
        subHeader: `We're sorry, there was an issue. Tap Dismiss then pull-down to refresh screen or try again later.`,
        buttons: [
          {
            text: 'Dismiss',
            role: 'cancel',
          },
        ],
      });
    }
    if (error.status && error.body && error.body.error) {
      return this.alertCtrl.create({
        header: `${error.status} error`,
        subHeader: error.body.error,
        buttons: [
          {
            text: 'Dismiss',
            role: 'cancel',
          },
        ],
      });
    }
    return this.alertCtrl.create({
      header: 'An error occurred',
      subHeader: 'Retry or contact support',
      message: error.message,
      buttons: [
        {
          text: 'Dismiss',
          role: 'cancel',
        },
      ],
    });
  }

  private async showError(error) {
    const alert = await this.getAlertForError(error);
    await alert.present();
  }
}
