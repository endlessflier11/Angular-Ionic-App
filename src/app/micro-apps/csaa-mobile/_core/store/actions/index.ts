export * from './config.actions';
export * from './metadata.actions';
export * from './service-locator.actions';
export * from './auth.actions';
export * from './fetch.actions';
export * from './policy.actions';
export * from './customer.actions';
export * from './payment.actions';
export * from './claim.actions';
export * from './user-setting.actions';
