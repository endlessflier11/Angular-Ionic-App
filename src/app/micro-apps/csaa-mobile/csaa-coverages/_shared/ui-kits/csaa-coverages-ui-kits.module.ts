import { NgModule } from '@angular/core';
import { CsaaPupCoveragesCardComponent } from './csaa-pup-coverages-card/csaa-pup-coverages-card.component';
import { CsaaGeneralCoveragesCardComponent } from './csaa-general-coverages-card/csaa-general-coverages-card.component';
import { CsaaDriversCoveragesCardComponent } from './csaa-drivers-coverages-card/csaa-drivers-coverages-card.component';
import { CsaaHomeCoveragesCardComponent } from './csaa-home-coverages-card/csaa-home-coverages-card.component';
import { CsaaVehiclesCoveragesCardComponent } from './csaa-vehicles-coverages-card/csaa-vehicles-coverages-card.component';
import { UiKitsModule } from '../../../_core/ui-kits/ui-kits.module';
import { IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';
import { CsaaCoverageCardItemComponent } from './csaa-coverage-card-item/csaa-coverage-card-item.component';
import {
  CsaaCoveragesVehicleCoveredCardComponent
} from './csaa-vehicle-coverages-covered-card/csaa-coverages-vehicle-covered-card.component';
import {
  CsaaCoveragesVehicleNotCoveredCardComponent
} from './csaa-vehicle-coverages-not-covered-card/csaa-coverages-vehicle-not-covered-card.component';
import { CsaaDriversCoveragesContentV1Component } from './csaa-drivers-coverages-content-v1/csaa-drivers-coverages-content-v1.component';
import { CsaaDriversCoveragesContentV2Component } from './csaa-drivers-coverages-content-v2/csaa-drivers-coverages-content-v2.component';

const SHARED_COMPONENTS = [
  CsaaCoverageCardItemComponent,
  CsaaPupCoveragesCardComponent,
  CsaaGeneralCoveragesCardComponent,
  CsaaDriversCoveragesCardComponent,
  CsaaDriversCoveragesContentV1Component,
  CsaaHomeCoveragesCardComponent,
  CsaaVehiclesCoveragesCardComponent,
  CsaaCoveragesVehicleCoveredCardComponent,
  CsaaCoveragesVehicleNotCoveredCardComponent,
  CsaaDriversCoveragesContentV2Component,
];

@NgModule({
  declarations: [...SHARED_COMPONENTS],
  imports: [CommonModule, IonicModule, UiKitsModule],
  exports: [...SHARED_COMPONENTS],
})
export class CsaaCoveragesUiKitsModule {}
