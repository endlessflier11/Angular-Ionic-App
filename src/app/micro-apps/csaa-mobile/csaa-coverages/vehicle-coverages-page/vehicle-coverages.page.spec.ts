/* eslint-disable @typescript-eslint/naming-convention */
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { IonicModule, NavController } from '@ionic/angular';
import { of } from 'rxjs';
import { CallService } from '../../_core/services/call.service';
import { CustomErrorHandler } from '../../_core/shared/custom-error-handler';
import { PolicyDocumentHelper } from '../../_core/shared/policy-document.helper';
import { UiKitsModule } from '../../_core/ui-kits/ui-kits.module';
import { CsaaVehicleCoveragesPage } from './vehicle-coverages.page';

import { CsaaCoveragesUiKitsModule } from '../_shared/ui-kits/csaa-coverages-ui-kits.module';
import { ActivatedRoute } from '@angular/router';

import { GlobalStateService } from '../../_core/services';
import {
  AUTH_STATE_FIXTURE_MOCK,
  CONFIG_STATE_FIXTURE_MOCK,
  CONTACT_INFO_STATE_FIXTURE_MOCK,
  CustomerResponseMock,
  PageTestingModule,
  PoliciesResponseMock,
} from '../../../../../testing';
import { AppEndpointsEnum } from '../../_core/interfaces';
import { HttpTestingController } from '@angular/common/http/testing';
import { Store } from '@ngxs/store';
import { CsaaAppInjector } from '../../csaa-app.injector';
import { Injector } from '@angular/core';
import { trackRequest } from '../../_core/operators/track-request.operator';
import { ContactInfoAction } from '../../_core/store/actions/contact-info.actions';

const flushMockResponses = (
  httpTestingController: HttpTestingController,
  MOCK: { customer: any; policies: any }
) => {
  httpTestingController
    .match(AppEndpointsEnum[AppEndpointsEnum.customerSearch])
    .forEach((req) => req.flush(MOCK.customer.toJson()));

  httpTestingController
    .match(AppEndpointsEnum[AppEndpointsEnum.policies])
    .forEach((req) => req.flush(MOCK.policies.toJson()));
};

describe('CsaaVehicleCoveragesPage', () => {
  let MOCK: { customer: any; policies: any };
  let component: CsaaVehicleCoveragesPage;
  let fixture: ComponentFixture<CsaaVehicleCoveragesPage>;
  let httpTestingController: HttpTestingController;

  beforeEach(() => {
    MOCK = {
      customer: CustomerResponseMock.create(),
      policies: PoliciesResponseMock.create(),
    };

    const [DBPolicyRef] = MOCK.policies.toJson();
    const [DBVehicleRef] = DBPolicyRef.vehicles;
    const PARAM_MAP = new Map();
    PARAM_MAP.set('policyNumber', DBPolicyRef.policyNumber);
    PARAM_MAP.set('vehicleId', DBVehicleRef.vehicleIdentifier);
    try {
      TestBed.configureTestingModule({
        declarations: [CsaaVehicleCoveragesPage],
        providers: [
          {
            provide: ActivatedRoute,
            useValue: {
              paramMap: of(PARAM_MAP),
            },
          },
          { provide: NavController, useValue: { push: jest.fn() } },
          {
            provide: CallService,
            useValue: {
              call: jest.fn(),
              getServiceNumber: jest.fn(() => '12345'),
              getEmergencyNumber: jest.fn(() => '45678'),
              getClaimsNumber: jest.fn(() => '1234'),
            },
          },
          { provide: CustomErrorHandler, useValue: { handleError: jest.fn() } },
          {
            provide: PolicyDocumentHelper,
            useValue: {
              openDocument: jest.fn(),
            },
          },
          {
            provide: GlobalStateService,
            useValue: {
              getIsStandalone: jest.fn(() => false),
              getRegistrationId: jest.fn(),
            },
          },
        ],
        imports: [
          IonicModule,
          UiKitsModule,
          CsaaCoveragesUiKitsModule,
          PageTestingModule.withConfig({
            providesStorage: true,
            providesPlatform: true,
            providesConfig: true,
            providesRouter: true,
            providesAnalytics: true,
          }),
        ],
      }).compileComponents();

      const store = TestBed.inject(Store);
      store.reset({
        csaa_app: {
          ...store.snapshot().csaa_app,
          csaa_config: CONFIG_STATE_FIXTURE_MOCK,
          csaa_auth: AUTH_STATE_FIXTURE_MOCK,
          csaa_contactInfo: CONTACT_INFO_STATE_FIXTURE_MOCK,
        },
      });

      CsaaAppInjector.injector = TestBed.inject(Injector);
      httpTestingController = TestBed.inject(HttpTestingController);
      fixture = TestBed.createComponent(CsaaVehicleCoveragesPage);
      component = fixture.componentInstance;

      // Todo: A more elegant way to set fetch status to true without api call
      of(null).pipe(trackRequest(ContactInfoAction.LoadContacts)).subscribe();
      fixture.detectChanges();
    } catch (error) {
      console.error(error);
      fail(error.message);
    }
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    component.ionViewWillEnter();
    flushMockResponses(httpTestingController, MOCK);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
