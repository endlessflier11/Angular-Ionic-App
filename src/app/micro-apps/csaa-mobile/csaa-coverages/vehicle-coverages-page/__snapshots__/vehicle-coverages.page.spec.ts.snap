// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CsaaVehicleCoveragesPage should match snapshot 1`] = `
<csaa-vehicle-coverages
  __ngContext__={[Function LRootView]}
  activatedRoute={[Function Object]}
  backButtonSubscription={[Function Observable]}
  coverages={[Function Array]}
  currentTheme=""
  errorHandler={[Function Object]}
  isAndroidPlatform="false"
  isPbeEligible$={[Function Observable]}
  loading="false"
  notCoverages={[Function Array]}
  platform={[Function PlatformMock]}
  policy$={[Function Observable]}
  policyDocumentHelper={[Function Object]}
  policyNumber={[Function String]}
  routerService={[Function RouterMockService]}
  store={[Function Store]}
  subsink={[Function SubSink]}
  title={[Function String]}
  vehicle={[Function Object]}
  vehicleId={[Function String]}
>
  <ion-header
    class="csaa"
  >
    <ion-toolbar>
      <ion-buttons
        slot="start"
      >
        <csaa-back-button
          text="Back"
        >
          <ion-button>
            <ion-icon
              name="chevron-back-outline"
              slot="start"
            />
             Back
          </ion-button>
        </csaa-back-button>
      </ion-buttons>
      <ion-buttons
        slot="end"
      >
        <ion-button>
          Call Service
        </ion-button>
      </ion-buttons>
    </ion-toolbar>
  </ion-header><ion-content
    class="csaa-item-list csaa"
  >
    <ion-refresher
      slot="fixed"
    >
      <ion-refresher-content />
    </ion-refresher>
    <h1
      class="csaa-content-title"
    >
      2018 Volkswagen Atlas
    </h1>
    <csaa-vehicle-coverages-covered-card>
      <ion-card
        class="csaa-card coverage-list"
      >
        <ion-card-header>
          <ion-item>
            <ion-card-title>
              Covered
            </ion-card-title>
          </ion-item>
        </ion-card-header>
        <ion-card-content
          class="csaa-item-list"
        >
          <ion-list>
            <ion-item>
              <csaa-coverage-card-item>
                <div
                  class="csaa-coverage-card-item"
                >
                  <h4
                    class="title"
                  >
                    Comprehensive
                  </h4>
                  <div
                    class="unopened-card"
                  >
                    <p
                      class="limit-info"
                    >
                      <span>
                        $500
                      </span>
                       Deductible 
                    </p>
                  </div>
                </div>
              </csaa-coverage-card-item>
            </ion-item>
            <ion-item>
              <csaa-coverage-card-item>
                <div
                  class="csaa-coverage-card-item"
                >
                  <h4
                    class="title"
                  >
                    Collision
                  </h4>
                  <div
                    class="unopened-card"
                  >
                    <p
                      class="limit-info"
                    >
                      <span>
                        $500
                      </span>
                       Deductible 
                    </p>
                  </div>
                </div>
              </csaa-coverage-card-item>
            </ion-item>
            <ion-item>
              <csaa-coverage-card-item>
                <div
                  class="csaa-coverage-card-item"
                >
                  <h4
                    class="title"
                  >
                    Special Equipment
                  </h4>
                  <div
                    class="unopened-card"
                  >
                    <p
                      class="limit-info"
                    >
                      <span>
                        $1,500
                      </span>
                       Per Disablement 
                    </p>
                  </div>
                </div>
              </csaa-coverage-card-item>
            </ion-item>
          </ion-list>
          <ion-button
            expand="full"
          >
            View Declarations
          </ion-button>
        </ion-card-content>
      </ion-card>
    </csaa-vehicle-coverages-covered-card>
    <csaa-vehicle-coverages-not-covered-card>
      <ion-card
        class="csaa-card coverage-list csaa-item-list"
      >
        <ion-card-header>
          <ion-card-title>
            Not Covered
          </ion-card-title>
        </ion-card-header>
        <ion-card-content
          class="csaa-item-list"
        >
          <ion-list>
            <ion-item>
              <csaa-coverage-card-item>
                <div
                  class="csaa-coverage-card-item"
                >
                  <h4
                    class="title"
                  >
                    Medical Payments
                  </h4>
                  <div
                    class="unopened-card"
                  >
                    <p
                      class="limit-info"
                    >
                       Covers medical expenses 
                    </p>
                  </div>
                </div>
              </csaa-coverage-card-item>
            </ion-item>
            <ion-item>
              <csaa-coverage-card-item>
                <div
                  class="csaa-coverage-card-item"
                >
                  <h4
                    class="title"
                  >
                    Loan/Lease Protection
                  </h4>
                  <div
                    class="unopened-card"
                  >
                    <p
                      class="limit-info"
                    >
                       Covers your car 
                    </p>
                  </div>
                </div>
              </csaa-coverage-card-item>
            </ion-item>
            <ion-item>
              <csaa-coverage-card-item>
                <div
                  class="csaa-coverage-card-item"
                >
                  <h4
                    class="title"
                  >
                    Rental Reimbursement
                  </h4>
                  <div
                    class="unopened-card"
                  >
                    <p
                      class="limit-info"
                    >
                       Covers rental costs 
                    </p>
                  </div>
                </div>
              </csaa-coverage-card-item>
            </ion-item>
            <ion-item>
              <csaa-coverage-card-item>
                <div
                  class="csaa-coverage-card-item"
                >
                  <h4
                    class="title"
                  >
                    Towing & Labor
                  </h4>
                  <div
                    class="unopened-card"
                  >
                    <p
                      class="limit-info"
                    >
                       Covers your car 
                    </p>
                  </div>
                </div>
              </csaa-coverage-card-item>
            </ion-item>
            <ion-item>
              <csaa-coverage-card-item>
                <div
                  class="csaa-coverage-card-item"
                >
                  <h4
                    class="title"
                  >
                    New Car Added Protection
                  </h4>
                  <div
                    class="unopened-card"
                  >
                    <p
                      class="limit-info"
                    >
                       Covers your car 
                    </p>
                  </div>
                </div>
              </csaa-coverage-card-item>
            </ion-item>
            <ion-item>
              <csaa-coverage-card-item>
                <div
                  class="csaa-coverage-card-item"
                >
                  <h4
                    class="title"
                  >
                    Ride-sharing (TNC)
                  </h4>
                  <div
                    class="unopened-card"
                  >
                    <p
                      class="limit-info"
                    >
                       Covers specified driver and vehicle 
                    </p>
                  </div>
                </div>
              </csaa-coverage-card-item>
            </ion-item>
          </ion-list>
        </ion-card-content>
      </ion-card>
    </csaa-vehicle-coverages-not-covered-card>
  </ion-content><csaa-working-hours
    class="csaa"
    servicetype="CUSTOMER"
  >
    <div
      class="csaa-working-hours hidden"
    >
      <ion-card
        class="card card-md"
      >
        <ion-card-header>
          Service Hours
        </ion-card-header>
        <ion-card-content
          class="card-content card-content-md"
        >
          <ion-list>
            <ion-item>
              <div
                class="csaa-working-hours-item"
              >
                <span
                  class="days"
                >
                  Mon-Fri
                </span>
                <span
                  class="hours"
                >
                  8am-5pm
                </span>
              </div>
              <div
                class="csaa-working-hours-item"
              >
                <span
                  class="days"
                >
                  Mon-Fri
                </span>
                <span
                  class="hours"
                >
                  8am-5pm
                </span>
              </div>
            </ion-item>
          </ion-list>
          <ion-button
            expand="full"
          >
             Call Service 
          </ion-button>
        </ion-card-content>
      </ion-card>
    </div>
  </csaa-working-hours>
</csaa-vehicle-coverages>
`;
